code/script.js

```diff
@@ -1,19 +1,108 @@
-// Start here
 
 // Step 1 - Welcome and introduction
-// Your code goes here
+
 alert(
-  `Welcome to our Javascript Pizzeria. Ready to Start? - Click 'OK' to begin.`
+  `Welcome to my JavaScript Pizzeria. Ready to Start? - Click 'OK' to begin.`
 )
 
+//UserName
+const userName = prompt("Please enter your name")
+
+alert(`Welcome ${userName}! Good to see you! Let's take a look at our different food options, shall we?`);
+
+
 // Step 2 - Food choice
-// Your code goes here
+const foodChoice = prompt(`What would you like today?
+1 - Pizza
+2 - Salad
+3 - Pasta
+Please enter the number of your choice.`);
+
+let selectedFood = "";
+
+if (foodChoice === "1") {
+  selectedFood = "Pizza";
+} else if (foodChoice === "2") {
+  selectedFood = "Salad";
+} else if (foodChoice === "3") {
+  selectedFood = "Pasta";
+} else {
+  alert("Invalid option or input! Please select a valid number :)");
+  process.exit(1);
+}
+
+alert(`You've chosen ${selectedFood}. Great choice!`);
 
 // Step 3 - Subtype choice
-// Your code goes here
+
+let foodType = "";
+let foodName = "";
+//foodType = parseInt(foodType); Do I need this? Don't think I do, unless I set a price to different foods.
+
+switch (selectedFood) {
+  case "Pizza":
+    foodType = prompt(`Choose your type of pizza:
+    1 - Margarita
+    2 - Hawaii
+    3 - Glava Special
+    Enter the number of your preferred pizza:`);
+    foodName = ["Margarita", "Hawaii", "Glava Special"];
+    break;
+
+  case "Salad":
+    foodType = prompt(`Choose your type of salad:
+      1 - Ceasar salad
+      2 - Greek salad
+      3 - Roadkill special
+      Enter the number of your preferred salad:`);
+    foodName = ["Ceasar salad", "Greek salad", "Roadkill special"];
+    break;
+
+  case "Pasta":
+    foodType = prompt(`Choose your type of pasta:
+      1 - Pasta Bolognese
+      2 - Pasta Alfredo
+      3 - Pasta la vista, baby! (surprice pasta!)
+      Enter the number of your preferred pasta:`);
+    foodName = ["Pasta Bolognese", "Pasta Alfredo", "Pasta la vista, baby! (surprice pasta!)"];
+    break;
+
+  default:
+    alert("Invalid option or input. Please select a valid number :)");
+    process.exit;
+  //the default doesnt work. Whyyyyyyyyyyy?!?
```

Comment:
Working perfectly. You could leave the former faulty code here, so you could still see, what was wrong before and adding a comment. Like: code -> now working because...